src/components/utils/Markdown.tsx

```diff
@@ -0,0 +1,165 @@
+import { useState, useEffect, useMemo } from "react";
+import ReactMarkdown from "react-markdown";
+import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
+import { coy } from "react-syntax-highlighter/dist/esm/styles/prism";
+import remarkGfm from "remark-gfm";
+import styled from "@emotion/styled";
+import { Button, Box } from "@chakra-ui/react";
+import { CopyIcon } from "@chakra-ui/icons";
+import { Row } from "../../utils/chakra";
+import { copySnippetToClipboard } from "../../utils/clipboard";
+
+// Required to display accurate font sizes for markdown elements.
+const StyledMarkdownWrapper = styled(Box)`
+  h1,
+  h2,
+  h3,
+  h4,
+  h5,
+  h6 {
+    font-size: inherit;
+    font-weight: 500;
+  }
+
+  blockquote {
+    margin: revert;
+  }
+
+  ol,
+  ul {
+    margin-left: 20px;
+  }
+  h1 {
+    font-size: 2em;
+  }
+
+  h2 {
+    font-size: 1.5em;
+  }
+
+  h3 {
+    font-size: 1.17em;
+  }
+
+  h4 {
+    font-size: 1em;
+  }
+
+  h5 {
+    font-size: 0.83em;
+  }
+
+  h6 {
+    font-size: 0.67em;
+  }
+
+  hr {
+    background-color: #000000;
+    height: 2px;
+    border: 0px;
+  }
+`;
+
+export interface MarkdownProps {
+  text: string;
+}
+
+const TitleBar = ({ language, code }: { language?: string; code: string }) => {
+  return (
+    <Row
+      mainAxisAlignment="flex-start"
+      crossAxisAlignment="stretch"
+      expand
+      justifyContent="space-between"
+      alignItems="center"
+      px="10px"
+      py="5px"
+      backgroundColor="#f5f5f5"
+      borderBottom="1px solid #eee"
+      borderRadius="6px"
+    >
+      <Box>{language || "plaintext"}</Box>
+      <CopyCodeButton code={code} />
+    </Row>
+  );
+};
+
+const CopyCodeButton = ({ code }: { code: string }) => {
+  const [copied, setCopied] = useState(false);
+
+  const handleCopyButtonClick = async (e: MouseEvent) => {
+    e.stopPropagation(); // Prevent this from triggering edit mode in the parent.
+
+    if (await copySnippetToClipboard(code)) setCopied(true);
+  };
+
+  useEffect(() => {
+    if (copied) {
+      const timer = setTimeout(() => setCopied(false), 2000);
+      return () => clearTimeout(timer);
+    }
+  }, [copied]);
+
+  return (
+    <Button
+      // TODO fix this
+      // @ts-ignore
+      onClick={handleCopyButtonClick}
+      size="xs"
+      variant="ghost"
+      px="5px"
+      _hover={{ background: "#EEEEEE" }}
+    >
+      <CopyIcon boxSize={4} mr={1} /> {copied ? "Copied!" : "Copy Code"}
+    </Button>
+  );
+};
+
+export function Markdown({ text }: { text: string }) {
+  const markdown = useMemo(
+    () => (
+      <StyledMarkdownWrapper>
+        <ReactMarkdown
+          remarkPlugins={[remarkGfm]} // supports github flavored markdown.
```

Comment:
this basically adds like tables and one or two other pretty irrelevant features? kinda in favor of just not including for bundle size sake, but if we want tables we should probably style them appropriately, they look kinda plain rn hehe
 
<img width="461" alt="Screenshot 2023-04-03 at 1 30 40 PM" src="https://user-images.githubusercontent.com/26209401/229620777-a254f445-05e2-4c69-9fad-f73b1031319a.png">

Reply:
I'm ok with not including it!

Reply:
sg, lets do that for now. can always add in the future if folks ask